x/profiles/keeper/msg_server_chain_link.go

```diff
@@ -25,6 +25,11 @@ func (k msgServer) LinkChainAccount(goCtx context.Context, msg *types.MsgLinkCha
 		return nil, err
 	}
 
+	// The first chain link of each chain for the owner will be set as default external address
+	if !k.HasDefaultExternalAddress(ctx, link.User, link.ChainConfig.Name) {
+		k.SaveDefaultExternalAddress(ctx, link.User, link.ChainConfig.Name, srcAddrData.GetValue())
+	}
```

Comment:
I think it's better to move this inside the keeper method that stores the chain link (`SaveChainLink`) instead of here

Reply:
From my side, `SaveDefaultExternalAddress` is different thing from `SaveChainLink`. 
Moving `SaveDefaultExternalAddress` to `SaveChainLink` will influence the behavior of `ImportGenesis`, say, the first chain link will be set as default even it is not `DefaultExnternalAddress`, so we need to update to the real one after.
As a result, I decided to let them separately.

Reply:
From my side, SaveDefaultExternalAddress is different thing from SaveChainLink.
Moving SaveDefaultExternalAddress to SaveChainLink will influence the behavior of ImportGenesis, say, the first chain link will be set as default even it is not DefaultExnternalAddress, so we need to update to the real one after.
As a result, I decided to let them separately.

Reply:
The problem with this approach (i.e. having the default address set inside the message handler) is that if we implement different ways of creating chain links (maybe using InterChain Accounts), then we will have to replicate these lines inside other parts of the code. I think it's much easier to: 
1. Move these lines inside the `SaveDefaultExternalAddress`
2. Make sure that inside `ImportGenesis` we first save the various chain links and *after that* we save the default links data. 

Thanks to simulation tests (particularly Import-Export tests) we should also be sure that the data is imported correctly

Reply:
Got you, it makes more sense.